code/Model/SiteTree.php

```diff
@@ -2844,6 +2871,52 @@ public function isHomePage(): bool
         return $this->URLSegment === RootURLController::get_homepage_link();
     }
 
+    public function getDefaultSearchContext()
+    {
+        return SiteTreeSearchContext::create(
+            static::class,
+            $this->scaffoldSearchFields(),
+            $this->defaultSearchFilters()
+        );
+    }
+
+    public function searchableFields()
+    {
+        $fields = parent::searchableFields();
+        // Push `FilterClass` to the top of the fields list
+        $fields = array_merge([
+            'FilterClass' => [
+                'filter' => '',
+                'general' => false,
+                'title' => _t(__CLASS__ . '.Filter_Status', 'Page status'),
+                'field' => DropdownField::class,
+            ]
+        ], $fields);
+        return $fields;
+    }
+
+    public function scaffoldSearchFields($_params = null)
+    {
+        $fields = parent::scaffoldSearchFields($_params);
+
+        // Update "Status" sources
+        $filters = CMSSiteTreeFilter::get_all_filters();
+        // Remove 'All records' as we set that to empty/default value
+        unset($filters[CMSSiteTreeFilter_Search::class]);
+        $fields->dataFieldByName('FilterClass')?->setSource($filters)->setEmptyString(CMSSiteTreeFilter_Search::singleton()->title());
```

Comment:
```suggestion
        $filterClass = $fields->dataFieldByName('FilterClass');
        if ($filterClass) {
            $title = CMSSiteTreeFilter_Search::singleton()->title();
            $filterClass->setSource($filters)->setEmptyString($title);
        }
```
I don't like everything on one line, it's hard to read.

Reply:
Done